scripts/layouts_tool/layout.h

```diff
@@ -0,0 +1,32 @@
+#ifndef LAYOUT_H
+#define LAYOUT_H
+
+#include <QString>
+#include <QX11Info>
+#include <QStringList>
+#include "defs.h"
+
+struct KbdKeyChar {
+    char16_t character;
+    bool is_dead;
+};
+
+typedef const KbdKeyChar (*KeyboardLayoutPointer)[2];
+
+class Layout {
+public:
+    Layout(QString varName, QString name, KeyboardLayoutPointer pData);
+    Layout(QString varName, QString name);
+    virtual ~Layout();
+
+    QStringList generateCode();
```

Comment:
This function generates the code for this layout. [This](https://github.com/Tomasito665/mixxx/blob/68716e1a5b22e0c430b57229245741fbf6bf2884/src/controllers/keyboard/layouts.cpp#L380-L437), for example. I couldn't come up with a better name, any suggestions are welcome :)

Anyway, I moved this function to `LayoutsFileHandler`, renamed it to `generateCodeForLayout` added some comments.